core/base/planarGraphLayout/MergeTreeVisu.h

```diff
@@ -0,0 +1,2060 @@
+  "-------------------------" << std::endl; std::cout << branchNodeOrigin << " _
+  " << previousBranchNodeOrigin << std::endl; printTuple(first);
+  printTuple(second); if(isConflictingBounds(first, second)) std::cout << i << "
+  _ " << j << " conflict" << std::endl;
+        }
+      }*/
+
+    } // end while
+
+    // ----- Correction of important/non-important pairs gap
+    // Get real gap
+    double realImportantPairsGap = std::numeric_limits<double>::lowest();
+    /*if(customImportantPairsSpacing)
+      realImportantPairsGap = importantPairsGap;
+    else{
+      for(auto leaf : leaves)
+        queueCrossing.push(leaf);
+      while(!queueCrossing.empty()){
+        ftm::idNode node = queueCrossing.front();
+        queueCrossing.pop();
+        ftm::idNode nodeOrigin = tree->getNode(node)->getOrigin();
+        //if(isRoot(tree, nodeOrigin)) continue; // TODO manage root gap and gap
+    for the others
+
+        bool isBranchImportant = isImportantPair<dataType>(tree, nodeOrigin,
+    ImportantPairs); if(not isBranchImportant) continue;
+
+        double gap = retVec[treeSimplexId[node]*2] -
+    std::get<0>(allBranchBounds[nodeOrigin]); realImportantPairsGap =
+    std::max(gap, realImportantPairsGap);
+      }
+      realImportantPairsGap *= 1.05;
+    }*/
+    realImportantPairsGap = importantPairsGap;
+
+    // Shift given real gap
+    for(auto leaf : leaves)
+      queueCrossing.push(leaf);
+    while(!queueCrossing.empty()) {
+      ftm::idNode node = queueCrossing.front();
+      queueCrossing.pop();
+      ftm::idNode nodeOrigin = tree->getNode(node)->getOrigin();
+
+      bool isBranchImportant
+        = isImportantPair<dataType>(tree, nodeOrigin, ImportantPairs);
+      if(not isBranchImportant)
+        continue;
+
+      for(unsigned int i = 0; i < allBranchOrigins[nodeOrigin].size(); ++i) {
+        ftm::idNode branchNodeOrigin = allBranchOrigins[nodeOrigin][i];
+        bool isSubBranchImportant
+          = isImportantPair<dataType>(tree, branchNodeOrigin, ImportantPairs);
+        double shift = 0;
+        if(not isSubBranchImportant) {
+          double gap = retVec[treeSimplexId[node] * 2]
+                       - std::get<0>(allBranchBounds[nodeOrigin]);
+          shift = -(realImportantPairsGap - gap) * NonImportantPairsProximity;
+        } else {
+          shift = -(
+            importantPairsGap
+            - realImportantPairsGap); // TODO multi shift depending on conflict
+        }
+        shiftBranchBounds(retVec, treeSimplexId, branching, allBranchBounds,
+                          allBranchOrigins[branchNodeOrigin], tree,
+                          branchNodeOrigin, shift);
+      }
+    }
+
+    if(verbose > 0) {
+      std::stringstream ss;
+      ss << "AVOID CROSSING  = " << t_avoid.getElapsedTime();
+      printMsg(ss.str(), debug::Priority::VERBOSE);
+    }
+
+    if(verbose > 0)
+      printMsg("================================", debug::Priority::VERBOSE);
```

Comment:
same remark as in PR #608 regarding separators.

Reply:
Ok I will fix this in both PR, thanks!